### PR TITLE
More examples for the docs

### PR DESCRIPTION
**Pull request by kolibril13**
Just some more examples for the example section of the docs

### CODE REVIEW

File: docs/source/examples/text.rst [hunk 1]
```diff
@@ -9,107 +9,6 @@ Text
             text = Text('Hello world').scale(3)
             self.add(text)
```
Comment (kolibril13) [inline docs/source/examples/text.rst:11]: Here I would like to make a reference to https://manimce--525.org.readthedocs.build/en/525/reference/manim.mobject.svg.text_mobject.Text.html#manim.mobject.svg.text_mobject.Text , something like `:class:`~.Text`
@behackl , could you maybe once show me the exact syntax for that? :)

**Comment by leotrs** [comment]
@kolibril13 you assigned Ben and me to this PR. I *think* you may have meant to ask us for a review. Note these two are different things. You can ask for a review at the top right (just above the "asignees" part).

File: manim/mobject/svg/tex_mobject.py [hunk 2]
```diff
@@ -152,6 +152,19 @@ def organize_submobjects_left_to_right(self):
 
 
 class MathTex(SingleStringMathTex):
+    """Some
+    Examples
```
Comment (leotrs) [inline manim/mobject/svg/tex_mobject.py:156]:
```suggestion
    """Some
    
    Examples
```

Comment (leotrs) [reply manim/mobject/svg/tex_mobject.py:156]: Please also add a docstring :)

File: manim/mobject/svg/tex_mobject.py [hunk 3]
```diff
@@ -152,6 +152,19 @@ def organize_submobjects_left_to_right(self):
...
+    --------
+    .. manim:: Formula1
+        :save_last_frame:
+
+        class Formula1(Scene):
+            def construct(self):
+                t = MathTex(r"\int_a^b f'(x) dx = f(b)- f(a)")
+                self.add(t)
+                self.wait(1)
```
Comment (leotrs) [inline manim/mobject/svg/tex_mobject.py:165]:
```suggestion
```

**Review by leotrs — COMMENTED** [review]
Beautiful! Just left a few minor comments.

File: docs/source/examples/annotations.rst [hunk 4]
```diff
@@ -15,3 +15,40 @@ Annotations
...
+            self.add(numberplane, dot, arrow, origin_text, tip_text)
+
+.. manim:: ExampleArrow2
+    :quality: medium
+    :save_last_frame:
+
+    from manim.mobject.geometry import ArrowTriangleTip, ArrowSquareTip, ArrowCircleTip
+    class ExampleArrow2(Scene):
+        def construct(self):
+            a11 = Arrow(np.array([-2, 3, 0]), np.array([2, 3, 0]))
```
[the remark this answers is not in the document]
Comment (kolibril13) [reply docs/source/examples/annotations.rst:38]: Yep, It is possible

File: docs/source/examples/annotations.rst [hunk 5]
```diff
@@ -15,3 +15,40 @@ Annotations
...
+            a12 = Arrow(np.array([-2, 2, 0]), np.array([2, 2, 0]),
+                        tip_shape=ArrowTriangleTip,
+                        tip_style={'fill_opacity': 0, 'stroke_width': 3})
```
[the remark this answers is not in the document]
Comment (kolibril13) [reply docs/source/examples/annotations.rst]: I have changed this now a bit, see here for the latest version of the example:
https://manimce--525.org.readthedocs.build/en/525/examples/annotations.html

File: docs/source/examples/annotations.rst [hunk 6]
```diff
@@ -15,3 +15,38 @@ Annotations
...
+            a11 = Arrow(start=[-2, 2, 0], end=[2, 2, 0], tip_shape=ArrowTriangleTip)
+            a12 = Arrow(start=[-2, 1, 0], end=[2, 1, 0],
+                        tip_shape=ArrowTriangleTip,
+                        tip_style={'fill_opacity': 1, 'stroke_width': 3})
+            a21 = Arrow(start=[-2, 0, 0], end=[2, 0, 0], tip_shape=ArrowSquareTip)
+            a22 = Arrow([-2, -1, 0], [2, -1, 0], tip_shape=ArrowSquareTip,
+                        tip_style={'fill_opacity': 1, 'stroke_width': 3})
+            a31 = Arrow([-2, -2, 0], [2, -2, 0], tip_shape=ArrowCircleTip)
+            a32 = Arrow([-2, -3, 0], [2, -3, 0], tip_shape=ArrowCircleTip,
+                        tip_style={'fill_opacity': 1, 'stroke_width': 3})
```
Comment (behackl) [inline docs/source/examples/annotations.rst:48]: There are separate classes for filled vs. unfilled tips, this is how it currently works:

```suggestion
    from manim.mobject.geometry import ArrowTriangleTip, ArrowSquareTip, ArrowSquareFilledTip,\
                                       ArrowCircleTip, ArrowCircleFilledTip
    class ExampleArrowTips(Scene):
        def construct(self):
            a00 = Arrow(start=[-2, 3, 0], end=[2, 3, 0], color=YELLOW)
            a11 = Arrow(start=[-2, 2, 0], end=[2, 2, 0], tip_shape=ArrowTriangleTip)
            a12 = Arrow(start=[-2, 1, 0], end=[2, 1, 0])
            a21 = Arrow(start=[-2, 0, 0], end=[2, 0, 0], tip_shape=ArrowSquareTip)
            a22 = Arrow([-2, -1, 0], [2, -1, 0], tip_shape=ArrowSquareFilledTip)
            a31 = Arrow([-2, -2, 0], [2, -2, 0], tip_shape=ArrowCircleTip)
            a32 = Arrow([-2, -3, 0], [2, -3, 0], tip_shape=ArrowCircleFilledTip)
```

(I think it makes sense to document custom settings via `tip_style` as well, but I'd rather something like that in the reference manual.)

File: docs/source/examples/text.rst (hunk 1 above)
Comment (behackl) [reply docs/source/examples/text.rst:11]: I think adding references like that is a really good idea for now! You almost had it right, simply say

```
This is an example that illustrates how to use the :class:`~.Text` class.
```

(The surrounding text is just to illustrate how to use these references within a paragraph; you probably want to replace it by something that actually makes sense.)

**Review by behackl — CHANGES_REQUESTED** [review]
This looks really good to me! Please consider my suggestion regarding the filled arrow tips - and I am not sure whether the comment you added at some point regarding referencing something from the reference manual is still open; let me know if you intend to do something with that.

File: docs/source/examples/annotations.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (behackl) [reply docs/source/examples/annotations.rst]: :+1:, good idea. Removing `np.array` makes it more readable.

File: docs/source/examples/annotations.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (behackl) [reply docs/source/examples/annotations.rst]: @leotrs Yes: https://github.com/ManimCommunity/manim/pull/525#discussion_r503145651

File: docs/source/examples/annotations.rst (hunk 6 above)
Comment (kolibril13) [reply docs/source/examples/annotations.rst]: cool. nice!
tip_style in the reference makes sence

File: docs/source/examples/text.rst (hunk 1 above)
Comment (kolibril13) [reply docs/source/examples/text.rst:11]: Cool.
Just added it :)

**Review by behackl — APPROVED** [review]
I thik this is good to go. Great additions!

File: manim/mobject/svg/tex_mobject.py [hunk 7]
```diff
@@ -152,8 +152,11 @@ def organize_submobjects_left_to_right(self):
 
 
 class MathTex(SingleStringMathTex):
-    """Some
-    Examples
+    """
+    A class for displaying mathematical formulas with Latex syntax.
+
+
+    Some Examples
```
[the remark this answers is not in the document]
Comment (kolibril13) [reply manim/mobject/svg/tex_mobject.py]: ok, nice!

**Review by leotrs — APPROVED** [review]

**Pull request merged**